### PR TITLE
chore: update stylelint to latest

### PR DESCRIPTION
**Pull request by jwbrandon**
### This PR will...
update the stylelint dependency which we have fallen way behind on.

### Why is this Pull Request needed?
The version of stylelint that we are using has a ton of npm audit issues.

#### Addresses Issue(s):

JW8-###

### Checklist
- [ ] Jenkins builds and unit tests are passing
- [ ] I have reviewed the automated results

### CODE REVIEW

File: .stylelintrc [hunk 1]
```diff
@@ -22,7 +23,7 @@
 
         "length-zero-no-unit": true,
 
-        "unit-blacklist": ["pt"],
```
Comment (jwbrandon) [inline .stylelintrc:25]: rule was renamed

File: .stylelintrc [hunk 2]
```diff
@@ -88,7 +89,7 @@
 
         "comment-no-empty": true,
         "comment-whitespace-inside": "always",
-        "comment-word-blacklist": ["/^TODO:/"],
```
Comment (jwbrandon) [inline .stylelintrc:91]: rule was renamed

File: .stylelintrc [hunk 3]
```diff
@@ -1,4 +1,5 @@
 {
+    "customSyntax": "postcss-less",
```
Comment (jwbrandon) [inline .stylelintrc:2]: stylelint no longer provides syntaxes for anthing but vanilla css as of version 14. We have to provide a syntax.

File: package.json [hunk 4]
```diff
@@ -12,11 +12,13 @@
     "@babel/helper-module-imports": "7.0.0",
     "@babel/preset-env": "7.3.1",
     "@babel/preset-typescript": "7.9.0",
+    "@lodder/grunt-postcss": "^3.1.1",
```
Comment (jwbrandon) [inline package.json:15]: grunt-postcss fell way behind and is now a readonly repository. Someone forked it and updated it to work with postcss 8.

File: package.json [hunk 5]
```diff
@@ -12,11 +12,13 @@
...
     "@types/webpack-env": "^1.15.1",
     "@typescript-eslint/eslint-plugin": "2.11.0",
     "@typescript-eslint/eslint-plugin-tslint": "2.11.0",
     "@typescript-eslint/parser": "2.11.0",
     "async": "2.6.0",
+    "autoprefixer": "^10.4.0",
```
Comment (jwbrandon) [inline package.json:21]: This used to be provided by `grunt-postcss` but due to postcss v8 changes we need to have it locally which seems correct in any case.

File: package.json [hunk 6]
```diff
@@ -58,15 +59,17 @@
     "marked": "0.7.0",
     "mocha": "5.2.0",
     "node-libs-browser": "2.1.0",
+    "postcss": "^8.4.4",
     "postcss-easing-gradients": "3.0.1",
-    "postcss-loader": "3.0.0",
-    "postcss-safe-parser": "3.0.1",
+    "postcss-less": "^5.0.0",
+    "postcss-loader": "^4.0.0",
```
Comment (jwbrandon) [inline package.json:65]: had to update postcss-loader to `v4` even though latest is `v6` because versions greater than 5 require webpack 5.

**Comment by jwbrandon** [comment]
test this please

**Comment by jwbrandon** [comment]
test this please

File: src/css/shared-imports/mixins/position.less [hunk 7]
```diff
@@ -108,6 +116,7 @@
  * Simultaneously set the top and right position of the element.
  *
  */
+/* stylelint-disable-next-line selector-class-pattern */
```
Comment (zetagame) [inline src/css/shared-imports/mixins/position.less:119]: We should just disable this rule

**Pull request merged**